### PR TITLE
refactor(migrations): gradle-lite

### PR DESCRIPTION
**Pull request by pret-a-porter**
## Changes:

One more migration from previously huge PR https://github.com/renovatebot/renovate/pull/12957

## Context:

1. https://github.com/renovatebot/renovate/issues/11459
2. Part of https://github.com/renovatebot/renovate/pull/12957 

## Documentation

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

File: lib/config/migrations/custom/gradle-lite-migration.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import is from '@sindresorhus/is';
+import { mergeChildConfig } from '../../utils';
```
Comment (viceice) [inline lib/config/migrations/custom/gradle-lite-migration.ts:2]: this causes strict null check issues :-/

Comment (pret-a-porter) [reply lib/config/migrations/custom/gradle-lite-migration.ts:2]: To be honest I need help with this issue, I spend few hours and could not find workaround. Excluding `gradle-lite-migration.ts` did not help. ts-ignore comment as well

Comment (viceice) [reply lib/config/migrations/custom/gradle-lite-migration.ts:2]: we should probalby shift this migration to a later time, as it needs strict null checks for almost all code or new exclusions in `tsconfig.strict.json` (don't like that, as we need to add previously working files).

Comment (rarkins) [reply lib/config/migrations/custom/gradle-lite-migration.ts:2]: Converted to draft

Comment (viceice) [reply lib/config/migrations/custom/gradle-lite-migration.ts:2]: should be unblocked now

File: lib/config/migrations/custom/gradle-lite-migration.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+export class GradleLiteMigration extends AbstractMigration {
+  override readonly deprecated = true;
+  override readonly propertyName = 'gradle-lite';
+
+  override run(value: unknown): void {
+    const gradle = this.get('gradle');
+
+    if (is.nonEmptyObject(value)) {
+      const newGradle = mergeChildConfig(gradle ?? {}, value);
```
Comment (viceice) [inline lib/config/migrations/custom/gradle-lite-migration.ts:13]: You probably need to cast `value`

**Comment by viceice** [comment]
I'm confused about the failing docker tests

File: lib/config/migrations/custom/gradle-lite-migration.ts (hunk 2 above)
Comment (pret-a-porter) [reply lib/config/migrations/custom/gradle-lite-migration.ts]: Yes, it helped, but now build failed and some strange issues with code coverage.

**Comment by rarkins** [comment]
Closing due to staleness

**Pull request closed without merging**